utils/src/storage.rs

```diff
@@ -0,0 +1,123 @@
+use std::{collections::HashMap, convert::Infallible};
```

Comment:
`utils` is an abbreviation for utilities. Does it make sense to put storage here? Or, we should create an individual module to represent the storage layer.

Reply:
moved to xline, my origin idea was to share this storage abstraction with curp, so I put it inside utils